src/binary-writer.cc

```diff
@@ -863,27 +894,54 @@ void BinaryWriter::WriteExpr(const Func* func, const Expr* expr) {
       has_data_segment_instruction_ = true;
       break;
     }
-    case ExprType::MemoryFill:
+    case ExprType::MemoryFill: {
+      Index memidx =
+          module_->GetMemoryIndex(cast<MemoryFillExpr>(expr)->memidx);
       WriteOpcode(stream_, Opcode::MemoryFill);
-      WriteU32Leb128(stream_, 0, "memory.fill reserved");
+      if (options_.features.multi_memory_enabled()) {
+        WriteU32Leb128(stream_, memidx, "memory.fill memidx");
+      } else {
+        WriteU32Leb128(stream_, memidx, "memory.fill reserved");
```

Comment:
I think think we need this if/else here.  Just `WriteU32Leb128(stream_, memidx, "memory.fill memidx");` should be fine.  Same below..

Reply:
This was kept for the same reason as OnOpcodeUint32Uint32Uint32.

Reply:
This doesn't make any sense to me since the final argument here is just the debug string which wabt prints.     I think its fine if this changes all the `memory.fill reserved` to `memory.fill memidx` in the test output if that is what you are worried about.

At the very least we should replace this with `WriteU32Leb128(stream_, memidx, options_.features.multi_memory_enabled() ? "memory.fill memidx" : "memory.fill reserved")` I think.

Reply:
Here is an example of the output:

> test/dump/current-memory.txt                              
>   STDERR MISMATCH:
>   --- expected
>   +++ actual
>   @@ -32,7 +32,7 @@
>    000001b: 00                                        ; func body size (guess)
>    000001c: 00                                        ; local decl count
>    000001d: 3f                                        ; memory.size
>   -000001e: 00                                        ; memory.size reserved
>   +000001e: 00                                        ; memory.size memidx
>    000001f: 0b                                        ; end
>    000001b: 04                                        ; FIXUP func body size
>    0000019: 06                                        ; FIXUP section size

We could just rebase the test output.